packages/aws-cdk-migration/lib/rewrite.ts

```diff
@@ -102,12 +112,29 @@ const EXEMPTIONS = new Set([
   '@aws-cdk/cloudformation-diff',
 ]);
 
-function updatedLocationOf(modulePath: string, options: RewriteOptions): string | undefined {
+function updatedLocationOf(modulePath: string, options: RewriteOptions, importedElements?: ts.NodeArray<ts.ImportSpecifier>): string | undefined {
   const customModulePath = options.customModules?.[modulePath];
   if (customModulePath) {
+    let awsCdkLibLocation = undefined;
+    importedElements?.forEach(e => {
+      if (e.name.text.startsWith('Cfn') || e.propertyName?.text.startsWith('Cfn')) {
+        // This is an L1 import, so don't return the customModulePath (which is the alpha module).
+        // Return the relevant aws-cdk-lib location.
+        awsCdkLibLocation = `aws-cdk-lib/${modulePath.substring(9)}`;
+      }
+    });
+    if (awsCdkLibLocation) {
+      return awsCdkLibLocation;
+    }
     return customModulePath;
   }
 
+  if (options.rewriteCfnImports && modulePath.endsWith('generated') && !modulePath.match(/canned-metric|augmentations/)) {
```

Comment:
minor: what happens if we add a new `xxx.generated.ts`. Either it should get excluded by default like canned and augmentations or an error thrown so that this code can be updated. Probably the former.

But the code here will actually rewrite. Feels like your matching be more inclusive than exclusive.

Reply:
Good point, will update to exclude anything besides `servicename.generated`. 

Since the current pattern is to use `-` to add more types of generated files (`ec2.generated`, `ec2-augmentations.generated`, `ec2-canned-metrics`), to do this, I could either
only rewrite L1 imports for anything that ends with `generated` and does not have a `-`. Or I could make sure it is an exact match with `servicename.generated`. For the latter, I would need to know the name of the current module/service by passing it in from `copy-files-removing-deps.ts`. This info is not currently available in the migration package as far as I know. 

I think the second option is better here. Let me know what you think!